### PR TITLE
Add extra construction recipes for furniture and terrain

### PR DESCRIPTION
**Pull request by Rail-Runner**
#### Summary

```SUMMARY: Content "Add extra construction recipes for furniture and terrain"```
#### Purpose of change

Adds many construction recipes, for the following map objects:

Furniture:

hay, desk, wardrobe, ~~lane guard, safe, dumpster, mailbox, glass front cabinet, heavy punching bag~~.

Terrain:

white grass, yellow waxed floor, removing paint from pavement and yellow waxed floor, ~~glass wall, glass door, reinforced glass door, bar door, window with metal bars,~~ railing, ~~glass railing, metal railing, concrete railing, sandbox, slide, monkey bars, backboard, little column,~~ covering manhole, ~~large metal support, small metal support, guard rail, smooth rock~~.

Some are mostly for decoration, some do have a practical use. I double-checked everything and made sure that smashing or deconstructing terrain and furniture specified above never produces more materials than amount used for their construction, and that it leaves correct terrain afterwards.

Changes to existing furniture and terrain:

1. Hay can now be deconstructed into rope and straw without using any tools or having to smash it.

2. Most doors that didn't leave frame after being destroyed now leave an appropriate one, so that a destroyed door could be replaced.

3. Monkey bars and slide now leave dirt instead of grass if smashed or deconstructed, to prevent creating grass out of nothing by constructing them.
#### Describe the solution

JSON editing.
#### Describe alternatives you've considered

Considered adding construction recipes to build rails (for railroad, since rail vehicles are a thing now), metal walls (like those found in labs) and concrete/metal bridges (the same as concrete and metal bridges found on map), but such complex constructions should probably be in their own PR. The idea is that despite their apparent complexity, those were somehow built by people before Cataclysm, with tools that they could bring there, so if materials and tools are available, it's likely possible to build them after Cataclysm too, even if it'd require high skill, lots of time and lots of materials. However, those factors would effectively restrict the player to repairing those structures if they're damaged, rather than building new ones from scratch, considering how many "tiles" one would have to build to do it.

Considered adding construction recipes to make some other terrain and furniture types, which would involve plastic and stone, but this needs a bit more time to figure out and also this PR is already quite large.

Considered adjusting a few construction recipes to be more consistent, but turns out that too many of them are a mess and it's probably a thing for another PR.
#### Additional context

All construction entries are grouped together in order to make resolving any possible conflicts easier.

### CODE REVIEW

**Comment by paulenka-aleh** [comment]
Sorry, welding a safe?.. The one that's with a dial, so heavy that it can't be moved and as secure as any pre-cataclysm one?

File: data/json/construction.json [hunk 1]
```diff
@@ -1624,6 +1624,449 @@
...
+    "post_terrain": "f_hay"
+  },
+  {
+    "type": "construction",
+    "description": "Build Desk",
+    "category": "FURN",
+    "required_skills": [ [ "fabrication", 1 ] ],
+    "time": 30,
+    "qualities": [ [ { "id": "HAMMER", "level": 2 } ] ],
+    "components": [ [ [ "2x4", 4 ] ], [ [ "nail", 10 ] ] ],
```
[the remark this answers is not in the document]
Comment (Rail-Runner) [reply data/json/construction.json]: Well, I was just looking at deconstruction/bash results while writing those recipes. Generally speaking, would it be okay to add extra 2x4's (or other items) to deconstruction and bash results if it turns out that the materials present there wouldn't be enough to make one?

**Comment by kevingranade** [comment]
The deconstruction recipes are only vaguely representative, which is ok for
deconstruction, but not for construction.
Adding reasonable construction recipes would justify increasing the yield
from deconstruction as well.

**Comment by Rail-Runner** [comment]
Done. Left only the most basic recipes. Desk and wardrobe now need more materials and are harder to make For desk, I assumed all-wood sliding drawers like those found on old desks, for simplicity. Wardrobe now needs a pipe to construct (that pipe on which you'd hang clothing).

**Comment by Rail-Runner** [comment]
> Sorry, welding a safe?.. The one that's with a dial, so heavy that it can't be moved and as secure as any pre-cataclysm one?

More like a "generic armored metal box with a door" than a safe, but likely the same thing for game purposes. Like a factory-made safe that was unlocked, it isn't actually lockable, but can be opened and closed. Also safes can be moved, though it does require player to be quite strong.

**Pull request merged**